### PR TITLE
fix Sometimes mentions list does not appear after typing @ in chat input filed

### LINKED ISSUE #15616: Sometimes mentions list does not appear after typing @ in chat input filed
Might be related to https://github.com/status-im/status-mobile/pull/15428 cc @qfrank 

Also we have e2e test which is failing from time to time because of current issue. 

Reproduced on Android (Samsung Galaxy A52, Android 12)

Try following steps to reproduce:

1. Open any chat or community channel
2. Start typing @ 
3. Clear input field
4. Background the app
5. Go back to the app
6. Start typing @ again
7. See if input mentions list has appeared
8. If input field has appeared try repeating steps 3-6 (see video below)

**Actual result:** sometimes mentions list does not appear

[Status-debug-logs (28).zip](https://github.com/status-im/status-mobile/files/11198906/Status-debug-logs.28.zip)

https://user-images.githubusercontent.com/97245802/231118095-ac92b000-2fa3-4361-b006-7a959a318639.mp4

### Additional Information

- Status version: nightly 
- Operating System: Android

[comment]: # (Please, add logs/notes if necessary)

### PR DESCRIPTION
**Pull request by qfrank**
fixes #15616 

status: ready

### CODE REVIEW

**Review by ibrkhalil — APPROVED** [review]

**Review by ilmotta — APPROVED** [review]

**Comment by pavloburykh** [comment]
@qfrank thanx for the PR. 

Sometimes am still facing the issue. Although there no clear steps of how to reproduce that. Below I attach video and logs from PR build. 

[Status-debug-logs (29).zip](https://github.com/status-im/status-mobile/files/11211281/Status-debug-logs.29.zip)

https://user-images.githubusercontent.com/97245802/231455240-1979d0c5-ab74-46da-9956-f352c80ed33f.mp4

**Comment by qfrank** [comment]
emm.. just probably found the cause 🤔

**Comment by churik** [comment]
@qfrank 
lmk when it is ready to be retested, thank you!

**Comment by qfrank** [comment]
> @qfrank lmk when it is ready to be retested, thank you!

ofc , but it's not ready yet 🤦‍♂️ . it will be ready to retest this evening or tomorrow morning @churik

**Comment by qfrank** [comment]
@pavloburykh @churik pls retry the latest build, thank you 🙂

File: src/status_im/chat/models/mentions.cljs [hunk 1]
```diff
@@ -91,38 +90,29 @@
...
-                                        :start         :Start
-                                        :end           :End})
-        params        [chat-id state]
-        method        "wakuext_chatMentionOnTextInput"]
-    (log/debug "[mentions] on-text-input" {:params params})
+(rf/defn on-change-text
+  {:events [:mention/on-change-text]}
+  [{:keys [db]} text]
+  (let [chat-id (:current-chat-id db)
+        params  [chat-id text]
```
Comment (flexsurfer) [inline src/status_im/chat/models/mentions.cljs:97]: no need in `let` anymore

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/chat/models/mentions.cljs (hunk 1 above)
Comment (qfrank) [reply src/status_im/chat/models/mentions.cljs:97]: but `params` was referenced in multi places, e.g. log/debug and on-error callback 🤔

Comment (flexsurfer) [reply src/status_im/chat/models/mentions.cljs:97]: ok

**Comment by churik** [comment]
@qfrank 
the issue is not reproducible anymore, let's wait for the e2e, and do not forget to squash my fixes while merging PR, thanks!

**Comment by churik** [comment]
@yevh-berdnyk 
tried selectors:
`//*[@content-desc='user-list']//*[@text='%s']/..//*[@content-desc='user-avatar']`
`//*[@content-desc='user-list']//*[@text='%s']/../..`

tried both `click_until_element_precense` and `click_until_element_absence` - no luck on clicking on element inside list.

Would you mind taking a look?

**Comment by churik** [comment]
@qfrank 
unfortunately, the bug is still there and it is easy to reproducible on Android emulators.
Video: https://app.eu-central-1.saucelabs.com/tests/41a23feca51644d0ab4e2efcd7c2eb6a#157
Logcat: 
[logcat (10).log](https://github.com/status-im/status-mobile/files/11270818/logcat.10.log) (from 23:08)

https://user-images.githubusercontent.com/4557972/233016255-5cf658bf-53c4-419b-b15d-f80ac251130b.mp4

And as in terms ofe2e tests it is getting worse here (in order to tap on an item I need to re-open the list several times) I cannot approve it for merging.

Perharps it is worth to wait until new composer is merged and then try this behaviour there.

**Comment by qfrank** [comment]
> Perharps it is worth to wait until new composer is merged and then try this behaviour there.

+1 @churik

**Comment by pavloburykh** [comment]
> Perharps it is worth to wait until new composer is merged and then try this behaviour there.

Also will need check behaviour when mention list does not appear after typing @ while editing message. 
Reproducible on latest nightly (Apr 24, 2023). 

https://user-images.githubusercontent.com/97245802/233958546-1000c1ed-358c-4909-8cc1-0ea1a2f0fcc5.mp4

**Comment by qfrank** [comment]
update: just rebased with [new composer](https://github.com/status-im/status-mobile/pull/15639)

**Comment by qfrank** [comment]
sorry... i made a mistake.. i force pushed and ignored commits that @churik made.. cc @pavloburykh

**Comment by qfrank** [comment]
just fixed a small issue, after user selected mention user from mention list, type @ won't popup mention list , it's ready to retest now

**Comment by churik** [comment]
@qfrank sorry for a long testing cycle this time!
Pr is ready to be merged.

**Review by churik — APPROVED** [review]

**Pull request merged**